MVCFinal/src/main/java/Controlador/ControladorPanelPoblacion.java

```diff
@@ -0,0 +1,150 @@
+		super(modelo, vista, controlador);
+		this.locales = this.getModelo().getConsultas().conseguirLocales(); }
+
+	public void mostrarPanel() {
+		this.total = 0;
+		this.panelPoblacion = makePanelPoblacion(this);
+		super.getVista().mostrarPanel(this.panelPoblacion); }
+	
+	public void borrarPlatosyProductosPanel() {
+		this.getModelo().getListaTemporal().limpiarLista();
+		this.getModelo().getListaTemporalPlatos().limpiarLista();
+		this.total = 0;	}
+	
+
+	public PanelPoblacion makePanelPoblacion(ControladorPanelPoblacion controladorPanelPoblacion) {
+		return new PanelPoblacion(controladorPanelPoblacion); }
+
+	public String leerNumTransBBDD() {
+		return String.valueOf(this.getModelo().getConsultas().leerNumTransBBDD()); }
+
+	public String[] conseguirLocales() {
+		String[] nombreLocales = new String[locales.size()];
+		int i = 0;
+		for (String[] array : locales) {
+			nombreLocales[i] = array[1];
+			i++; }
+		return nombreLocales;
+	}
+	public String conseguirActividadesLocal(int selectedIndex) {
+		return locales.get(selectedIndex)[2];}
+
+	public String[] cogerListaProductos() {
+		return this.getModelo().getListaProductos().getListaProductosString();}
+
+	public String devolverFechaHora() {
+		return this.getModelo().getFechaHoraSys(); }
+
+	public int conseguirStock(int indexSelected, String producto) {
+		return this.getModelo().getConsultas().obtenerStock(devolverNifLocal(indexSelected), producto); }
+
+	public boolean comprobarCampos(double total, String nif, String nombre, String apellido) {
+		return total > 0 && this.getModelo().validaciones.comprobarCamposString(nif, nombre, apellido); }
+
+	public String accionadoBotonEliminar(int pos, String eliminar) {
+		this.total = this.getModelo().funProd.funcionalidadeliminarProducto(pos, eliminar, this.total);
+		return String.valueOf(total); }
+
+	public void accionadoBotonAnadirAprovisionamiento(int cantidad, int indice, String nombre, int selectedIndex) throws SQLException {
+		double precioTotal = this.getModelo().getConsultasComprobaciones().consultaComprobarPrecio(nombre);
+		this.getModelo().insercionesActividades.insertarActividad(this.getModelo().getConsultas().leerNumTransBBDD(),
+				devolverFechaFormateada(this.getModelo().getFechaHoraSys()), 0, "aprovisionamiento",devolverNifLocal(selectedIndex));
+		this.getModelo().insercionesActividades
+				.insertarAprovisionamiento(this.getModelo().getConsultas().leerNumTransBBDD() - 1);
+		this.getModelo().getInserciones().insertarProductoActividad( this.getModelo().getConsultas().leerNumTransBBDD() - 1,
+				this.getModelo().getConsultas().obtenerCodigoAlimentoProducto(nombre), cantidad, precioTotal,devolverNifLocal(selectedIndex)); 
+		this.getModelo().insercionesActividades.ejecutarFuncion(this.getModelo().getConsultas().leerNumTransBBDD(),"aprovisionamiento");
+	
+	}
+
+	public void insercionDatosBbdd(int transaccion, String fecha, double totalOperacion, int selectedIndex,
```

Comment:
Este método es demasiado largo. Yo haría un método privado para cada caso.

Reply:
Yo también haría varios, pero lo hemos hecho así para intentar reducir las líneas y optimizar código, separando ese método en uno por cada funcionalidad (ticket,factura..) van a salirnos muchas líneas.

Reply:
La idea es que cuando un método hace muchas cosas es incorrecto por definición. Y, esto siempre debe prevalecer sobre el resto de cuestiones. La norma es que un método ha de tener una única funcionalidad. 
Lo que podéis hacer es trasladar a una nueva clase las funcionalidades de aprovisionamiento, pedido, comanda y factura. Y, que el método que tengamos en este controlador solo se dedique a gestionar que método habrá de llamarse de esa nueva clase.

Reply:
Lo hemos pasado a una clase llamada funcionalidad población la cual lo gestiona el llamando a métodos de la propia clase.